control/sum.go

```diff
@@ -34,25 +34,46 @@ func newSum(config BlockConfig, logger logging.Logger) (Block, error) {
 }
 
 func (b *sum) Next(ctx context.Context, x []*Signal, dt time.Duration) ([]*Signal, bool) {
-	if len(x) != len(b.operation) {
+	// sum blocks only support signals with the same number of inputs
+	if len(x)%2 != 0 {
 		return b.y, false
 	}
-	y := 0.0
-	for i := range x {
+	y := make([]float64, len(x)/2)
+	half := len(x) / 2
+
+	// loop through the first set of inputs and add or subract from the corresponding output
+	for i := 0; i < half; i++ {
 		op, ok := b.operation[x[i].name]
 		if !ok {
 			return b.y, false
 		}
 		switch op {
 		case addition:
-			y += x[i].GetSignalValueAt(0)
+			y[i] += x[i].GetSignalValueAt(0)
 		case subtraction:
-			y -= x[i].GetSignalValueAt(0)
-		default:
+			y[i] -= x[i].GetSignalValueAt(0)
+		}
+	}
+
+	// loop through the second set of inputs and add or subract from the corresponding output
+	for i := half; i < len(x); i++ {
+		op, ok := b.operation[x[i].name]
+		if !ok {
 			return b.y, false
 		}
+		switch op {
+		case addition:
+			y[i-half] += x[i].GetSignalValueAt(0)
+		case subtraction:
+			y[i-half] -= x[i].GetSignalValueAt(0)
+		}
 	}
-	b.y[0].SetSignalValueAt(0, y)
+
+	// loop through the output and set the signal
+	for i := range y {
+		b.y[i].SetSignalValueAt(0, y[i])
+	}
+
```

Comment:
okay tried to rework a little I think it's more general. basically will only support systems that have the same number of inputs from setpoint and endpoint, and that expects that many outputs. works for motor (1) and base (2) but should theoretically work for any number higher than that. you just couldn't do combining of signals it all has to stay separated